### PR TITLE
[clang/DependencyScanning/ModuleDepCollector] Refactor part of `makeCommonInvocationForModuleBuild` into its own function

### PR DESCRIPTION
**Pull request by akyrtzi**
The new function is about clearing out benign codegen options and can be applied for PCH invocations as well.

### CODE REVIEW

File: clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h [hunk 1]
```diff
@@ -308,6 +308,9 @@ class ModuleDepCollector final : public DependencyCollector {
                                 ModuleDeps &Deps);
 };
 
+/// Resets some options that introduce dependencies unnecessarily.
+void removeUnnecessaryDependencies(CompilerInvocation &CI, bool ForModuleBuild);
```
Comment (jansvoboda11) [inline clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h:312]:
```suggestion
void removeUnnecessaryFileArguments(CompilerInvocation &CI, bool ForModuleBuild);
```

Dependencies is a bit ambiguous here.

File: clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp [hunk 2]
```diff
@@ -154,6 +154,20 @@ void ModuleDepCollector::addOutputPaths(CowCompilerInvocation &CI,
...
+                                                 bool ForModuleBuild) {
+  if (CI.getFrontendOpts().ProgramAction == frontend::GeneratePCH ||
+      (ForModuleBuild && !CI.getLangOpts().ModulesCodegen)) {
+    CI.getCodeGenOpts().DebugCompilationDir.clear();
+    CI.getCodeGenOpts().CoverageCompilationDir.clear();
+    CI.getCodeGenOpts().CoverageDataFile.clear();
+    CI.getCodeGenOpts().CoverageNotesFile.clear();
+    CI.getCodeGenOpts().ProfileInstrumentUsePath.clear();
+    CI.getCodeGenOpts().SampleProfileFile.clear();
+    CI.getCodeGenOpts().ProfileRemappingFile.clear();
```
Comment (jansvoboda11) [inline clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp:167]: We inherit these from the translation unit, are their values usable for modules?

Comment (akyrtzi) [reply clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp:167]: I had an offline discussion with Jan about how to better factor-out this piece of code.

Comment (akyrtzi) [reply clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp]: See updated commit.

File: clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h (hunk 1 above)
Comment (akyrtzi) [reply clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h]: New name in updated commit.

**Review by jansvoboda11 — APPROVED** [review]
LGTM. I assume you'll call `resetBenignCodeGenOptions()` from `ModuleDepCollector::applyDiscoveredDependencies()` in a follow-up, non-NFC patch, right? (So that it applies to the primary invocation, i.e. the actual PCH.) Maybe leaving a FIXME behind would be nice, just to make the intent clear.

**Comment by akyrtzi** [comment]
> I assume you'll call resetBenignCodeGenOptions() from ModuleDepCollector::applyDiscoveredDependencies() in a follow-up, non-NFC patch, right?

@jansvoboda11 see updated commit, I made the change per your suggestion.

**Review by jansvoboda11 — APPROVED** [review]
Thanks! Still LGTM.

**Pull request merged**